C7Engine/AI/UnitAI/CombatAI.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections.Generic;
+using C7GameData;
+using C7GameData.AIData;
+using Serilog;
+
+namespace C7Engine.AI.UnitAI {
+	/// <summary>
+	/// A unit whose intended role is combat, other than purely point-defense combat.
+	///
+	/// This is likely to evolve significantly, and likely have sub-classes; this is the very first iteration.
+	/// This first iteration is focused on defeating barbarians.
+	/// </summary>
+	public class CombatAI : C7Engine.UnitAI {
+		private Tile target;
+
+		private ILogger log = Log.ForContext<DefenderAI>();
+
+		public bool PlayTurn(Player player, MapUnit unit) {
+			CombatAIData explorerData = (CombatAIData)unit.currentAIData;
+
+			while (true) {
```

Comment:
I'm not sure I follow why this is an infinite loop and it looks kind of dangerous. Are we trying to find where along the original path the unit is now? Would it make sense to represent the path as something like a stack or linked list and pop the next tile with each move?

Reply:
This definitely looked more confusing coming back to it after a week than right after I wrote it.  I don't remember why it has an infinite loop.  Best guess is it's supposed to move until it's out of movement points or it engages in combat, but it currently won't move more than one tile even if it has movement points for it.

Definitely needs changes!  Thanks for calling this out while I still remember some of why I wrote it the way I wrote it!

The path _is_ represented as a stack (TilePath.java), but that hasn't been fully integrated into the AI yet.

Reply:
This has been revamped.  Now it will use existing methods to follow its path to the barbarian camp.